imgui-SFML.cpp

```diff
@@ -666,17 +765,178 @@ void Update(sf::Window& window, sf::RenderTarget& target, sf::Time dt) {
 
         Update(s_currWindowCtx->touchPos, static_cast<sf::Vector2f>(target.getSize()), dt);
     } else {
-        Update(sf::Mouse::getPosition(window), static_cast<sf::Vector2f>(target.getSize()), dt);
+        if (ImGui::GetIO().ConfigFlags & ImGuiConfigFlags_ViewportsEnable)
+            Update(sf::Mouse::getPosition(), static_cast<sf::Vector2f>(target.getSize()), dt);
+        else
+            Update(sf::Mouse::getPosition(window), static_cast<sf::Vector2f>(target.getSize()), dt);
     }
 }
 
 void Update(const sf::Vector2i& mousePos, const sf::Vector2f& displaySize, sf::Time dt) {
     assert(s_currWindowCtx && "No current window is set - forgot to call ImGui::SFML::Init?");
-
     ImGuiIO& io = ImGui::GetIO();
+    
     io.DisplaySize = ImVec2(displaySize.x, displaySize.y);
     io.DeltaTime = dt.asSeconds();
 
+#ifdef VIEWPORTS_ENABLE
+    if (io.ConfigFlags & ImGuiConfigFlags_ViewportsEnable) {
+        SFML_UpdateMonitors();
+
+        for (auto& viewport : ImGui::GetPlatformIO().Viewports) {
+            WindowContext* wc = (WindowContext*)viewport->PlatformUserData;
```

Comment:
`ImGui::SFML::ProcessEvent` handles only main window events. Events for viewports' windows are not handled by it. Maybe, to avoid copy-pasting, `ImGui::SFML::ProcessEvent` could call internal function, for example `ProcessEvent(const Event&, WindowContext*)`, with `s_currWindowCtx`, and in this loop it would be called with corresponding viewport's window context?